File: server/DTOs/ProductDto.cs
namespace server.DTOs
{
    public class ProductDetailDto
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public DateTime CreationDate { get; set; }

        public int ProductCategoryId { get; set; }

        public int CreatorUserId {get;set;}

        public List<int> Categories { get; set; } = new();

        public List<CommentDto> Comments { get; set; } = new();
    }

    public class ProductListDto
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public DateTime CreationDate { get; set; }

        public int CreatorUserId { get; set; }

    }
}

File: client/src/DTOs/ProductDTO.ts
import { CommentDto } from './CommentDto';

export interface CategoryDto {
    id: number;
    name: string;
}

export interface ProductDetailDto {
    id: number;
    title: string;
    description: string;
    imageUrl: string | null;
    categoryId: number;
    creationDate: Date | null;
    creatorUserId: number;
    comments: Array<CommentDto>;
    productCategoryList: Array<string>;
    productCategoryIdList: Array<number>;
   
}

export interface ProductListDto {
    id: number;
    title: string;
    description: string;
    imageUrl: string | null;
    categoryId: number;
    creationDate: Date | null;
    creatorUserId: number;

}

export interface NewProductDto {
    title: string;
    description: string;
    imageUrl: string | null;
    productCategoryIdList: Array<number>;
}


